lib/src/polygon_to_line.dart

```diff
@@ -0,0 +1,69 @@
+import '../helpers.dart';
+
+/// Converts a [Polygon] to [LineString] or [MultiLineString] or a [MultiPolygon] to a
+/// [FeatureCollection] of [LineString] or [MultiLineString].
+/// Returns [FeatureCollection] or [Feature<LineString>] or [Feature<MultiLinestring>]
+/// example:
+/// ```dart
+/// var poly = Polygon(coordinates:
+///  [
+///   [
+///   Position.of([125, -30]),
+///   Position.of([145, -30]),
+///   Position.of([145, -20]),
+///   Position.of([125, -20]),
+///   Position.of([125, -30])
+///   ]
+/// ]);
+/// var line = polygonToLine(poly);
+/// //addToMap
+/// var addToMap = [line];
+/// ```
+polygonToLine(GeoJSONObject poly, {Map<String, dynamic>? properties}) {
+  var geom = poly is Feature ? poly.geometry : poly;
+
+  (poly is Feature)
+      ? properties = poly.properties
+      : properties = <String, dynamic>{};
+
+  if (geom is Polygon) {
+    return _polygonToLine(poly, properties: properties);
+  } else if (geom is MultiPolygon) {
+    return _multiPolygonToLine(poly, properties: properties);
+  } else {
+    throw Exception("invalid poly");
+  }
+}
+
+Feature _polygonToLine(GeoJSONObject poly, {Map<String, dynamic>? properties}) {
```

Comment:
why are you not giving this the parameter type `Polygon` and pass `geom` inside `polygonToLine` directly to it?